### ISSUE TITLE
Implement Noise Models

### ISSUE THREAD

**Issue by ProfFan** [borglab/SwiftFusion#51]
Currently we are assuming a Unitary noise model in all out factor graph implementations. Need to add one to the factors so we can solve more complicated problems :)

TODO:
- [ ] Add rudimentary noise model (Gaussian)
- [ ] Change the GFG and NLFG to accept additional noise model
- [ ] Update the examples

**Comment by dellaert** [comment]
I would argue this is less important at this time. Timeline/milestones doc coming soon.

**Comment by marcrasi** [comment]
I think this would be quite nice because it'll allow us to compare apples-to-apples the results and performance of our implementation to other implementations like https://github.com/borglab/gtsam/blob/develop/examples/Pose2SLAMExample_g2o.cpp.

**Comment by dellaert** [comment]
I agree it needs to be done at one point. But for the apples-apples, we can also just use unit noise models in GTSAM :-)

**Comment by ProfFan** [comment]
Per #69, this has become a necessity as all g2o benchmarks in 3D have noise models
